src/frequenz/sdk/_internal/_channels.py

```diff
@@ -29,6 +31,37 @@ def new_receiver(self, *, limit: int = 50) -> Receiver[T_co]:
         """
 
 
+class ReceiverFetcherWith(ReceiverFetcher[U_co], typing.Generic[T_co, U_co]):
```

Comment:
```suggestion
class ReceiverFetcherWith(typing.Generic[T_co, U_co]):
```

Inheriting from a protocol will make this a protocol, but we actually don't want this to be a protocol in itself, just a regular class that happens to implement the `ReceiverFetcher` protocol, right?

Reply:
true, it should be fine without deriving from it.  changed.